src/plugins/data/server/search/search_service.ts

```diff
@@ -328,36 +351,69 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
     return strategy.extend(id, keepAlive, options, deps);
   };
 
-  private getSearchStrategy = <
-    SearchStrategyRequest extends IKibanaSearchRequest = IEsSearchRequest,
-    SearchStrategyResponse extends IKibanaSearchResponse = IEsSearchResponse
-  >(
-    name: string = this.defaultSearchStrategyName
-  ): ISearchStrategy<SearchStrategyRequest, SearchStrategyResponse> => {
-    this.logger.debug(`Get strategy ${name}`);
-    const strategy = this.searchStrategies[name];
-    if (!strategy) {
-      throw new KbnServerError(`Search strategy ${name} not found`, 404);
+  private cancelSession = async (deps: SearchStrategyDependencies, sessionId: string) => {
+    const searchIdMapping = await deps.searchSessionsClient.getSearchIdMapping(sessionId);
+
+    for (const [searchId, strategyName] of searchIdMapping.entries()) {
```

Comment:
Update the status first, and don't fail `cancelSession` if cancelling one of the requests fails.

Reply:
Hmmm, why did we have to do the individual search request cancellations here and not inside the session service?
Lets talk about this?